src/callback_tracking.jl

```diff
@@ -101,6 +101,12 @@ function (f::TrackedAffect)(integrator, event_idx = nothing)
     end
 end
 
+# extend the DiffEqCallbacks.functioncalling_affect_initialize to fit TrackedAffect
+import DiffEqCallbacks: functioncalling_affect_initialize
+function functioncalling_affect_initialize(affect::TrackedAffect, u, t, integrator)
+    functioncalling_affect_initialize(affect.affect!, u, t, integrator)
+end
+
 function _track_callback(cb::DiscreteCallback, t, u, p, sensealg)
     correction = ImplicitCorrection(cb, t, u, p, sensealg)
     DiscreteCallback(cb.condition,
```

Comment:
Can you maybe add an MWE as a test set for this and (if we manage to do it) a custom initialization?

Reply:
adding the MWE to https://github.com/SciML/SciMLSensitivity.jl/tree/master/test/callbacks, so it gets tested and we don't miss updating it if there are changes e.g. to `TrackedAffect!` in the future.

Reply:
sure, no problem!